### DISCUSSION TITLE
Error! This MFA code is not valid

### DISCUSSION THREAD

**Discussion by AlexKalopsia** [firefly-iii/firefly-iii#6174] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Out of the blue, Firefly is not accepting my 2FA OTP codes. Whenever I input the code, I just get the error message `Error! This MFA code is not valid`.

I am on a single user setup and I have never had any issue with 2FA. The only thing is that I haven't logged in for the past 2 months, but that's about it. 

I also tried to update to the latest version, but that didn't help.

### Expected behaviour

Expected to accept my OTP code

### Debug information

Debug information generated at 2022-06-19 12:19:05 Europe/Stockholm for Firefly III version **5.7.9**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.9 |
| Firefly III API | 1.5.6 |
| Build | 196, 09-06-2022 18:30:26 CEST |
| DB version | 18 (exp. 18) |
| PHP | `8.0.17` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Stockholm |
| Browser TZ | Europe/Paris |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | info, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-04-01 00:00:00 |
| Session end | 2022-06-30 00:00:00 |
| Session first | 2015-01-01 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.124 Safari/537.36 Edg/102.0.1245.44 |

### Steps to reproduce

1. Reach Firefly
2. Add username and password
3. Insert OTP code

**Comment by JC5** [comment]
I'm sorry, I'm not sure what I can do for you. You could try resetting your 2FA settings? You'll need access to the database.

**Comment by AlexKalopsia** [comment]
No worries, could you guide me through resetting the 2FA settings?

**Comment by JC5** [comment]
https://docs.firefly-iii.org/firefly-iii/faq/other/#i-lost-my-2fa-token-generator-or-2fa-has-stopped-working

**Comment by AlexKalopsia** [comment]
Ok, so, for what it's worth, I reset the mfa secret, logged in, reenabled 2FA, but even during setup Firefly III doesn't accept the 6 digits code.

I tried both with QR code scan, as well as adding the secret manually. I am confused :S

**Comment by JC5** [comment]
Reset MFA, then clear all of your cookies just to be sure.

**Comment by AlexKalopsia** [comment, marked as answer]
Just as a final update here, the problem was with my Syno NAS. I noticed that other services also failed MFA, so it turns out that from the DSM settings, the clock sync was not working with the `time.google.com` servers. I changed it to `time.nist.gov`, setup MFA again, and it all worked again.

**Reply by JC5** [reply]
Damn, that's a mean one! I'm glad you found it. Thank you for getting back to me. I will extend the documentation to include this as a possible source of problems.

Thanks again!

**Reply by AlexKalopsia** [reply]
No problem, and yes it was sneaky, I wouldn't have checked the DSM time settings if I didn't have MFA issues on that other container. To be clear though, the DSM _did_ have a warning in the time panel saying that it couldn't retrieve data, I just never checked that in the DSM panel.
